### ISSUE TITLE
'Group by' function

### ISSUE THREAD

**Issue by jackosullivanoxford** [ebmdatalab/openpath-dash#101]
A couple of issues here:

**Comment by jackosullivanoxford** [comment]
1. When grouping by lab, a decile graph is generated, but how are we generating deciles if there are only three labs? Am I misunderstanding the unit of analysis here? 
<img width="1146" alt="Screen Shot 2019-12-03 at 5 46 52 PM" src="https://user-images.githubusercontent.com/35063801/70105823-cc2f2000-15f6-11ea-9241-2d02d0f13c9c.png">

**Comment by jackosullivanoxford** [comment]
2. When I select 'Group by': 'Test code' and alternate which lab is selected (under the tab 'Showing which labs') the decile graph is always the same, and is one line (not deciles). A further point on these graphs is the ESR data seems strange: the sharp increase and then sharp decline...
<img width="1146" alt="Screen Shot 2019-12-03 at 5 54 12 PM" src="https://user-images.githubusercontent.com/35063801/70105925-292ad600-15f7-11ea-95be-03b931f892a7.png">
<img width="1146" alt="Screen Shot 2019-12-03 at 5 54 05 PM" src="https://user-images.githubusercontent.com/35063801/70105927-292ad600-15f7-11ea-8f4e-8d202c17aa00.png">
<img width="1146" alt="Screen Shot 2019-12-03 at 5 53 55 PM" src="https://user-images.githubusercontent.com/35063801/70105928-292ad600-15f7-11ea-86ae-81bb1b65b186.png">

**Comment by jackosullivanoxford** [comment]
3. When I group by test code, and select >1 test, the decile graph changes from one line to a decile graph. I'm not sure how this is a decile graph if the unit of analysis is two tests?

<img width="1146" alt="Screen Shot 2019-12-03 at 5 58 12 PM" src="https://user-images.githubusercontent.com/35063801/70106018-6a22ea80-15f7-11ea-894f-c5067ebb4537.png">

**Comment by evansd** [comment]
Thanks @jackosullivanoxford, very useful feedback. You're right that the deciles thing is very confusing. (It's using linear interpolation to produce 10 lines even though there are fewer than 10 points.) We've already got a plan (issue #85) to not show deciles when we only have a small number of points so hopefully this problem should just go away.

It's also a bit confusing that the deciles don't change when filtering by CCG or Lab, but the plan here is to always show national deciles for comparison. I've made a note (issue #102) to hopefully make this clearer in the interface.

Finally, it looks like there's definitely a data issue with the ESR numbers. Seb's investigating that along with various other data issues that we've picked up.
